pkg/raft/testdata/async_storage_writes.txt

```diff
@@ -63,24 +62,27 @@ stabilize
   3->1 MsgVoteResp Term:1 Log:0/0
 > 1 receiving messages
   1->1 MsgVoteResp Term:1 Log:0/0
+  1->1 StorageAppendAck Mark:{Term:0 Index:0}
   INFO 1 received MsgVoteResp from 1 at term 1
   INFO 1 has received 1 MsgVoteResp votes and 0 vote rejections
 > 2 processing append thread
-  Processing:
   HardState {Term:1 Vote:1 Commit:10 Lead:0 LeadEpoch:0}
-  Responses:
+  OnSync:
   2->1 MsgVoteResp Term:1 Log:0/0
 > 3 processing append thread
-  Processing:
   HardState {Term:1 Vote:1 Commit:10 Lead:0 LeadEpoch:0}
-  Responses:
+  OnSync:
   3->1 MsgVoteResp Term:1 Log:0/0
 > 1 receiving messages
   2->1 MsgVoteResp Term:1 Log:0/0
   INFO 1 received MsgVoteResp from 2 at term 1
   INFO 1 has received 2 MsgVoteResp votes and 0 vote rejections
   INFO 1 became leader at term 1
   3->1 MsgVoteResp Term:1 Log:0/0
+> 2 receiving messages
```

Comment:
what changed here?

Reply:
Probably has to do with [this](https://github.com/cockroachdb/cockroach/pull/143705/files#diff-e732e1701c8a830413fb6c35b2b84e3f551956ea8bbd6b30936ce2d90b059765R67-R69) check. It should be `if NeedsAck() { append(ack) }` rather than `if MustSync() { append(ack) }`.

Will fix now.

Reply:
Fixed in the last commit.